src/remote/ticket.go

```diff
@@ -0,0 +1,68 @@
+package remote
+
+import (
+	"github.com/go-resty/resty/v2"
+	"keboola-as-code/src/client"
+	"keboola-as-code/src/model"
+	"sort"
+	"sync"
+)
+
+// TicketProvider generates new IDs and GUARANTEES that the IDs will be in the same order as the Request method was called
+type TicketProvider struct {
```

Comment:
Pre nove objekty, ktore najde `persist` cmd potrebujeme generovat nova IDs cez API.

Robi sa to pomocou parallelnych HTTP requestov (`pool`). 
No nie je garantovane poradie, ktory request dobehne prvý, atd.... v testoch to robilo problemy, ked bolo poradie vygenerovanych ID nahodne, preto som vytvoril strukturu `TicketProvider`.

`TicketProvider` garantuje:
- ze kazdy callback je zavolany v poradi, v akom bol definovany
- a kazdy callback dostane vygenerovane ID, od najmensieho po najvacsie

Reply:
bolo nutne robit generovanie IDs paralelne? Moja uvaha smeruje k tomu ze keby to bolo sekvencne tak by sa to nemuselo zozlozitit cez TicketProvider

Reply:
@kacurez nemuselo by sa to robit paralelne, ...ale takto je to rychlejsie, ... user moze kludne vytvorit naraz aj 20 konfiguracii, a je rozdiel ci bude cakat na `persist` command, `0.5s` alebo `7 sekund`.

Proste sa posle naraz parallelne napr. 20 requestov, ktore dobehnu priblizne v rovnakom case ako 1 samostatny request.

Reply:
jasne chapem. Nebolo by potom lepsie synchronizovat cez go channels nez cez mutexy? Pytam sa ciste len zo zvedavosti, implementaciu si neviem do detailov predstavit.

Reply:
`HTTP pool` interne pouziva `channels` (vod `pool.go`) ale tu som na to nevidel dovod:
- kod so zamkom je kratsi a prehladnejsi.
- zamok sa pouziva na zapis do spolocnej `slice`, ... tj. bude to zamknute par `uS`.
- kod s pouzitim `channels` by vyzadoval dalsiu `goroutine` ktora by z kanala citala a ukalada do `slice`.
- okrem toho by tam bola nejaka `waitgroup`, ktora by cakala na tu `goroutine`.
- teda nevidim ziaden prinos :) , iba negativa: dlhsi kod, menej prehladny kod, rovnako rychle.
- ak chces, mozes poslat PR a mozeme priamo porovnat kod

Reply:
ale dobra otazka :+1:

Reply:
jasne, ako som pisal tu implementaciu cez channels si zatial neviem do detailov predstavit, akurat vnimam ze channels sa zvyknu pouzit na podobne veci. To co pises dava zmysel! dik za dovysvetlenie ;)